### PR TITLE
CXP-1419 Create custom app

### PR DESCRIPTION
**Pull request by RiffVentura**
Moved and renamed code related to custom app creation into a dedicated namespace for Custom Apps
image
image
image
image

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 1]
```diff
@@ -301,6 +267,42 @@ akeneo_connectivity.connection:
...
+                        name: Define app name
+                        activate_url: Fill activate url
+                        callback_url: Fill callback url
+                credentials:
+                    title: Credentials
+                    client_id: Client id
+                    client_secret: Client secret
+                errors:
+                    name:
+                        min_length: The name of the test app is too short. It must have 3 characters or more.
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]:
```suggestion
                        min_length: The name of the custom app is too short. It must have 3 characters or more.
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 2]
```diff
@@ -301,6 +267,42 @@ akeneo_connectivity.connection:
...
+                        max_length: The name of the test app is too long. It must have 255 characters or less.
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]:
```suggestion
                        max_length: The name of the custom app is too long. It must have 255 characters or less.
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 3]
```diff
@@ -301,6 +267,42 @@ akeneo_connectivity.connection:
...
+                        not_blank: The name of the test app is required.
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]:
```suggestion
                        not_blank: The name of the custom app is required.
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 4]
```diff
@@ -301,6 +267,42 @@ akeneo_connectivity.connection:
...
+                    client_id:
+                        max_length: The client id of the test app is too long. It must have 36 characters or less.
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]:
```suggestion
                        max_length: The client id of the custom app is too long. It must have 36 characters or less.
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 5]
```diff
@@ -301,6 +267,42 @@ akeneo_connectivity.connection:
...
+                        not_blank: The client id is required.
+                    activate_url:
+                        max_length: The activate url of the test app is too long. It must have 255 characters or less.
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]:
```suggestion
                        max_length: The activate url of the custom app is too long. It must have 255 characters or less.
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 6]
```diff
@@ -301,6 +267,42 @@ akeneo_connectivity.connection:
...
+                        must_be_url: The activate url must be a valid URL.
+                        not_blank: The activate url is required.
+                    callback_url:
+                        max_length: The callback url of the test app is too long. It must have 255 characters or less.
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml]:
```suggestion
                        max_length: The callback url of the custom app is too long. It must have 255 characters or less.
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+import {mockFetchResponses, renderWithProviders} from '../../../test-utils';
+import {setLogger} from 'react-query';
+
+setLogger({
+    log: () => null,
+    warn: () => null,
+    error: () => null, // explicit error generation triggers react query to log the error
+});
+
+test('it creates the test app and returns credentials', async done => {
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:15]:
```suggestion
test('it creates the custom app and returns credentials', async done => {
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    mockFetchResponses({
+        akeneo_connectivity_connection_custom_apps_rest_create: {
+            json: {
+                clientId: 'testClientId',
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:19]:
```suggestion
                clientId: 'customClientId',
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+                clientSecret: 'testSecret',
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:20]:
```suggestion
                clientSecret: 'customSecret',
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+        return (
+            <div>
+                <h1>{data?.clientId}</h1>
+                <h1>{data?.clientSecret}</h1>
+                <button
+                    onClick={() => {
+                        mutate(
+                            {
+                                name: 'Custom app bynder',
+                                activateUrl: 'http://any_url.test',
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:38]:
```suggestion
                                activateUrl: 'http://any_url.custom',
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+                                callbackUrl: 'http://activate.test',
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:39]:
```suggestion
                                callbackUrl: 'http://activate.custom',
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+                        );
+                    }}
+                >
+                    mutate
+                </button>
+            </div>
+        );
+    };
+
+    renderWithProviders(<TestComponent />);
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:51]:
```suggestion
    renderWithProviders(<CustomComponent />);
```
? :thinking:

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+    await waitFor(() => screen.getByText('mutate'));
+
+    fireEvent.click(screen.getByText('mutate'));
+
+    await waitFor(() => screen.getByText('testClientId'));
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:57]:
```suggestion
    await waitFor(() => screen.getByText('customClientId'));
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+
+    expect(screen.getByText('testClientId')).toBeInTheDocument();
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:59]:
```suggestion
    expect(screen.getByText('customClientId')).toBeInTheDocument();
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+    expect(screen.getByText('testSecret')).toBeInTheDocument();
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:60]:
```suggestion
    expect(screen.getByText('customSecret')).toBeInTheDocument();
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+
+    expect(fetchMock).toBeCalledWith('akeneo_connectivity_connection_custom_apps_rest_create', {
+        body: '{"name":"Custom app bynder","activateUrl":"http://any_url.test","callbackUrl":"http://activate.test"}',
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:63]:
```suggestion
        body: '{"name":"Custom app bynder","activateUrl":"http://any_url.custom","callbackUrl":"http://activate.custom"}',
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+        headers: {
+            'Content-type': 'application/json',
+            'X-Requested-With': 'XMLHttpRequest',
+        },
+        method: 'POST',
+    });
+
+    expect(onSuccess).toBeCalledWith(
+        {
+            clientId: 'testClientId',
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:73]:
```suggestion
            clientId: 'customClientId',
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+            clientSecret: 'testSecret',
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:74]:
```suggestion
            clientSecret: 'customSecret',
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+                    {propertyPath: 'name', message: 'name_error'},
+                    {propertyPath: 'activateUrl', message: 'activate_url_error'},
+                    {propertyPath: 'callbackUrl', message: 'callback_url_error'},
+                ],
+            },
+            status: 422,
+        },
+    });
+
+    const TestComponent = () => {
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:97]:
```suggestion
    const CustomComponent = () => {
```

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+                        });
+                    }}
+                >
+                    mutate
+                </button>
+            </div>
+        );
+    };
+
+    renderWithProviders(<TestComponent />);
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/hooks/use-create-custom-app.test.tsx:121]:
```suggestion
    renderWithProviders(<CustomComponent />);
```

**Review by guillaumefer-akeneo — APPROVED** [review]

**Review by eoliveauakeneo — APPROVED** [review]

**Pull request merged**